stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -495,29 +518,124 @@ private static void setHttpStatusCodes(int statusCode, HttpCarbonMessage respons
      * @return Error struct
      */
     public static ErrorValue getError(String errMsg) {
-        MapValue<String, Object> httpErrorRecord = createHTTPErrorRecord();
+        MapValue<String, Object> httpErrorRecord = createHttpErrorDetailRecord(errMsg, null);
         httpErrorRecord.put(HTTP_ERROR_MESSAGE, errMsg);
         return BallerinaErrors.createError(HTTP_ERROR_CODE, httpErrorRecord);
     }
 
-    private static MapValue<String, Object> createHTTPErrorRecord() {
-        return BallerinaValues.createRecordValue(PROTOCOL_PACKAGE_HTTP, HTTP_ERROR_RECORD);
-    }
-
     /**
      * Get error struct from throwable.
      *
      * @param throwable Throwable representing the error.
      * @return Error struct
      */
     public static ErrorValue getError(Throwable throwable) {
+        if (throwable instanceof ClientConnectorException) {
+            return createHttpError(throwable);
+        }
         if (throwable.getMessage() == null) {
-            return getError(IO_EXCEPTION_OCCURED);
+            return createHttpError(IO_EXCEPTION_OCCURED);
+        } else {
+            return createHttpError(throwable.getMessage());
+        }
+    }
+
+    public static ErrorValue createHttpError(String errorMessage) {
+        HttpErrorType errorType = getErrorType(errorMessage);
+        return createHttpError(errorMessage, errorType);
+    }
+
+    public static ErrorValue createHttpError(Throwable throwable) {
```

Comment:
Why not write a set of overloaded methods for each exception type instead of checking instanceOf inside one method?

Reply:
Done